### ISSUE TITLE
TextShape visible property not working

### ISSUE THREAD

**Issue by nearnshaw** [decentraland/explorer#827]
I can't make a TextShape invisible
If I set visible = false, it remains visible

This simple example shows a text that should be invisible, but can be seen:

```
const myEntity2 = new Entity()
const myText2 = new TextShape('Goodbye World!')
myText2.visible = false
myEntity2.addComponent(myText2)
myEntity2.addComponent(new Transform({ position: new Vector3(8, 1, 8) }))
engine.addEntity(myEntity2)
```

[https://app.zenhub.com/files/220994476/dabe984b-e9d0-41b7-803f-5a2c3b27e5d0/download](https://app.zenhub.com/files/220994476/dabe984b-e9d0-41b7-803f-5a2c3b27e5d0/download)

**Comment by marcosnc** [comment]
@nchamo Can we fix this easily?

**Comment by nchamo** [comment]
Hey @marcosnc !

So I just a little digging and the `visible` property is sent to Unity. The problem is that Unity doesn't know that the property exists. The json received is deserialized into a [class](https://github.com/decentraland/explorer/blob/master/unity-client/Assets/Scripts/MainScripts/DCL/Components/TextShape/TextShape.cs#L11) that doesn't have that propery.

That means that either:
1. The SDK is exposing a property that shouldn't be exposed, or
2. The client is incorrectly unaware of such property. 

Whatever we (I don't know who should make the decision) decide, be might consider it an SDK bug, or a client bug.

**Comment by marcosnc** [comment]
I think we need to support the `visible` property, so it needs to be taken into account by Unity. @BrianAmadori what's the effort to process the property accordingly?

**Comment by AjimenezDCL** [comment]
The problem is `TextShape` in Unity not extending `BaseShape` where this property exists. This is problematic for various reasons:
- `TextShape` currently is a `BaseComponent` in Unity, that means that changing the parent class to `BaseShape` (which is a Disposable component) can get really messy.
- Another side effect of making `TextShape` a `BaseShape` are the properties `withCollision` and `isPointerBlocker`. After discussing it with Nico, these two are enabled by default in kernel, which means that the current deployed scenes using `TextShape` has these properties already set by default to `true`. If we implement said properties in Unity we might be introducing a gamebreaking change.

So from now on `TextShape` wont have these `withCollision` `isPointerBlocker` 


On a side note, this ticket will also fix #1392
